spec.md

```diff
@@ -4695,17 +4698,40 @@ File = {Statement | newline} eof .
 
 Statement = DefStmt | IfStmt | ForStmt | SimpleStmt .
 
-DefStmt = 'def' identifier '(' [Parameters [',']] ')' ':' Suite .
+DefStmt = 'def' identifier '(' [Parameters [',']] ')' ['->' TypeExpr ]  ':' Suite .
 
 Parameters = Parameter {',' Parameter}.
 
-Parameter  = identifier
-           | identifier '=' Test
+Parameter  = identifier [':' TypeExpr] ['=' Test]
            | '*'
-           | '*' identifier
-           | '**' identifier
+           | '*' identifier [':' TypeExpr]
+           | '**' identifier [':' TypeExpr]
            .
 
+LambdaParameters = LambdaParameter {',' LambdaParameter}.
+
+LambdaParameter  = identifier ['=' Test]
+                 | '*'
+                 | '*' identifier
+                 | '**' identifier
+                 .
+           
+TypeExpr = Type {'|' Type}.
+
+Type = identifier [TypeArguments].
```

Comment:
Nit: Use a more specific name so as not to confuse with the toplevel production (`TypeExpr`). Maybe `TypeAtom`?